tex/polyglossia.sty

```diff
@@ -729,11 +730,16 @@ That is almost guaranteed to cause problems.}
          \xpg@hook@setlanguage%
          \let\emp@langname\@undefined}%
       \aftergroup\xpg@pop@language%
+      \setkeys{xpg@langvariant}{#1}%
```

Comment:
FYI: This line made every language option different from _variant_ fail (e. g., calendar, locale, numerals, etc.  There are dozens of those for many languages).  I wrote commit https://github.com/reutenauer/polyglossia/commit/d2a34ff to fix that.  I expect that still does what you needed.

I haven’t made an upload to CTAN with the code in this pull request or #121 , but will very soon.